WordPress/gutenberg
# [RNMobile] Upgrade compile and target sdk version to Android API 31

Related PRs:
- **Parent PR in WordPress-Android:** https://github.com/wordpress-mobile/WordPress-Android/pull/17153
- https://github.com/wordpress-mobile/gutenberg-mobile/pull/5184
- https://github.com/wordpress-mobile/WordPress-Android/pull/17235



## What?


Upgrade `react-native-aztec`, `react-native-bridge` and `react-native-editor` projects to Android 12 (API 31).

## Why?


[Google Play requires apps to update to Android 12 (API level 31) before 31 November 1, 2022](https://support.google.com/googleplay/android-developer/answer/11926878?hl=en).

## How?


`compileSdkVersion` and `targetSdkVersion` properties of all Android projects have been updated to API 31.

Additionally, the main activity of the demo Android demo project is now exported. This is a behavior change required when using Android 12 related to safer component exporting ([reference](https://developer.android.com/about/versions/12/behavior-changes-12)).

The Android dependencies of forked repositories have been also updated to Android 12:
- [react-native-gesture-handler](https://github.com/wordpress-mobile/react-native-gesture-handler/pull/23)
- [react-native-reanimated](https://github.com/wordpress-mobile/react-native-reanimated/pull/19)
- [react-native-prompt-android](https://github.com/wordpress-mobile/react-native-prompt-android/pull/9)
- [react-native-linear-gradient](https://github.com/wordpress-mobile/react-native-linear-gradient/pull/7)
- [react-native-slider](https://github.com/wordpress-mobile/react-native-slider/pull/12)
- [react-native-video](https://github.com/wordpress-mobile/react-native-video/pull/20)

## Testing Instructions




This change doesn't affect a specific area, hence to test it we should basically smoke test the editor and check that works as expected.

## Screenshots or screencast 
N/A
